pallets/system/src/header_builder.rs

```diff
@@ -126,22 +124,39 @@ pub fn build_extension<M: Metrics>(
 	seed: Seed,
 	metrics: &M,
 ) -> HeaderExtension {
+	use once_cell::sync::Lazy;
+	static PMP: Lazy<kate::pmp::m1_blst::M1NoPrecomp> =
+		once_cell::sync::Lazy::new(|| kate::testnet::multiproof_params(256, 256));
 	use avail_core::header::extension::{v1, v2};
 
-	let (xts_layout, commitment, block_dims, _data_matrix) = kate::com::par_build_commitments(
-		block_length.rows,
-		block_length.cols,
-		block_length.chunk_size(),
-		app_extrinsics,
+	let timer = std::time::Instant::now();
+	let grid = kate::gridgen::EvaluationGrid::from_extrinsics(
+		app_extrinsics.to_vec(),
+		4,                                    //TODO: where should this minimum grid width be specified
+		block_length.cols.0.saturated_into(), // even if we run on a u16 target this is fine
+		block_length.rows.0.saturated_into(),
 		seed,
-		metrics,
 	)
-	.expect("Build commitments cannot fail .qed");
-	let app_lookup = DataLookup::from_id_and_len_iter(xts_layout.into_iter())
-		.expect("Extrinsic size cannot overflow .qed");
-	let rows = block_dims.rows().0.saturated_into::<u16>();
-	let cols = block_dims.cols().0.saturated_into::<u16>();
-
+	.expect("Grid construction cannot fail");
```

Comment:
nit: Pls use [QED](https://en.wikipedia.org/wiki/Q.E.D.) here or error handling.

Reply:
This isn't a QED. A QED is supposed to include a definite reason why it should work, always. This is just saying "hey if you find a way to panic commitment generation, this whole pipeline is totally screwed."